src/winutils.cc

```diff
@@ -0,0 +1,126 @@
+
+
+#include "winutils.h"
+#include "TlHelp32.h"
+#include <cassert>
+
+
+DWORD getppid() {
+    HANDLE hSnapshot = INVALID_HANDLE_VALUE;
+    PROCESSENTRY32 pe32;
+    DWORD ppid = 0, pid = GetCurrentProcessId();
+
+    hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
```

Comment:
Can't a significant portion of this function replaced by `CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, pid);`?